dashboard/state_aggregator.py

```diff
@@ -87,10 +116,75 @@ async def get_workers(self) -> dict:
             result[data["worker_id"]] = data
         return result
 
-    def _message_to_dict(self, *, message, fields_to_decode) -> dict:
+    async def get_tasks(self) -> dict:
+        # Need to wait until at least the head node stub is available.
+        await dashboard_utils.wait_for_stub(self._raylet_stub)
+
+        async def get_task_info(stub):
+            reply = await stub.GetTasksInfo(
+                node_manager_pb2.GetTasksInfoRequest(),
+                timeout=DEFAULT_RPC_TIMEOUT,
+            )
+            return reply
+
+        replies = await asyncio.gather(
+            *[get_task_info(stub) for stub in self._raylet_stub.values()]
+        )
+
+        result = defaultdict(dict)
+        for reply in replies:
+            tasks = reply.task_info_entries
+            for task in tasks:
+                data = self._message_to_dict(
+                    message=task,
+                    fields_to_decode=["task_id"],
+                )
+                logger.info(data)
+                data = filter_fields(data, TaskState)
+                result[data["task_id"]] = data
+        return result
+
+    async def get_objects(self) -> dict:
+        # Need to wait until at least the head node stub is available.
+        await dashboard_utils.wait_for_stub(self._raylet_stub)
+
+        async def get_object_info(stub):
+            reply = await stub.GetNodeStats(
+                node_manager_pb2.GetNodeStatsRequest(include_memory_info=True),
```

Comment:
might as well make this and `get_tasks` better structured for unit testing since it's new code. I would suggest doing something like I mentioned above:

```
class ObjectInfoProvider:
    async def get_objects(self): # abstract method
        pass
```
then have an implementation that deals with all of the grpc logic (including an `add_stub` interface)

Reply:
this way we can easily unit test the data munging logic